File: app/src/main/java/vn/loitp/app/activity/customviews/layout/draggableview/DraggableViewActivity.java
package vn.loitp.app.activity.customviews.layout.draggableview;

import android.os.Bundle;

import com.core.base.BaseFontActivity;
import com.core.utilities.LLog;
import com.views.layout.draggablepanel.DraggableListener;
import com.views.layout.draggablepanel.DraggableView;

import loitp.basemaster.R;

public class DraggableViewActivity extends BaseFontActivity {
    private DraggableView draggableView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        draggableView = (DraggableView) findViewById(R.id.draggable_view);

        draggableView.setClickToMaximizeEnabled(true);
        draggableView.setClickToMinimizeEnabled(true);
        draggableView.setHorizontalAlphaEffectEnabled(true);

        draggableView.setDraggableListener(new DraggableListener() {
            @Override
            public void onMaximized() {
                LLog.INSTANCE.d(getTAG(), "onMaximized");
            }

            @Override
            public void onMinimized() {
                LLog.INSTANCE.d(getTAG(), "onMinimized");
            }

            @Override
            public void onClosedToLeft() {
                LLog.INSTANCE.d(getTAG(), "onClosedToLeft");
            }

            @Override
            public void onClosedToRight() {
                LLog.INSTANCE.d(getTAG(), "onClosedToRight");
            }
        });
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_draggable_view;
    }
}

File: app/src/main/java/vn/loitp/app/activity/customviews/layout/draggableview/DraggableViewActivity.kt
package vn.loitp.app.activity.customviews.layout.draggableview

import android.os.Bundle
import com.annotation.IsFullScreen
import com.annotation.LayoutId
import com.annotation.LogTag
import com.core.base.BaseFontActivity
import com.views.layout.draggablepanel.DraggableListener
import kotlinx.android.synthetic.main.activity_draggable_view.*
import vn.loitp.app.R

@LayoutId(R.layout.activity_draggable_view)
@LogTag("DraggableViewActivity")
@IsFullScreen(false)
class DraggableViewActivity : BaseFontActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setupViews()
    }

    private fun setupViews() {
        draggableView.isClickToMaximizeEnabled = true
        draggableView.isClickToMinimizeEnabled = true
        draggableView.setHorizontalAlphaEffectEnabled(true)

        draggableView.setDraggableListener(object : DraggableListener {
            override fun onMaximized() {
                logD("onMaximized")
            }

            override fun onMinimized() {
                logD("onMinimized")
            }

            override fun onClosedToLeft() {
                logD("onClosedToLeft")
            }

            override fun onClosedToRight() {
                logD("onClosedToRight")
            }
        })
    }

}
